### PR TITLE
4.x - Added missing docblocks for traits, classes and interfaces

### PR DESCRIPTION
**Pull request by othercorey**
Found generating api docs.

### CODE REVIEW

File: src/View/Widget/YearWidget.php [hunk 1]
```diff
@@ -20,6 +20,12 @@
 use Cake\View\StringTemplate;
 use InvalidArgumentException;
 
+/**
+ * Input widget class for generating a year select box.
+ *
+ * This class is intended as an internal implementation detail
```
Comment (ADmad) [inline src/View/Widget/YearWidget.php]: This statement can be misleading. Someone might read it the class being an internal class which should not be extended, but that's not the case. Similar statement in other widgets should also be changed / removed.

Comment (othercorey) [reply src/View/Widget/YearWidget.php]: Let me know what you want it to say and I'll clean them up.

Comment (ADmad) [reply src/View/Widget/YearWidget.php]: I don't have a suggestion for rewording it, so I would just remove it.

Let's see what other's have to say.

Comment (markstory) [reply src/View/Widget/YearWidget.php]: The widget classes aren't entirely implementation details. If someone wanted to re-use this to make standalone year widgets they could. Perhaps something like

> This class can be used to generate standalone select boxes for choosing calendar years.

Comment (othercorey) [reply src/View/Widget/YearWidget.php]: I added that with a note about being used in FormHelper.  See if that works.

**Pull request merged**